pkg/virt-controller/watch/vmi.go

```diff
@@ -590,13 +622,23 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 		c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, SuccessfulCreatePodReason, "Created virtual machine pod %s", pod.Name)
 		return nil
+	} else {
+		if isTempPod(pod) && !isWaitForFirstConsumer {
+			// DV is no longer in WaitForFirstConsumer phase, means that tempPod has done its job and should be deleted
+			err := c.clientset.CoreV1().Pods(vmi.Namespace).Delete(pod.Name, &v1.DeleteOptions{})
```

Comment:
we need to have a delete expectation set here.

Reply:
ok